### PR TITLE
Merge swss and syncd into single service

### PR DESCRIPTION
**Pull request by marian-pritsak**
Current implementation of swss and syncd causes a lot
of problems in terms of dependencies and synchronization.
Instead of handling them in separate services, we now
start and stop them both as a single entity.

Signed-off-by: marian-pritsak <marianp@mellanox.com>

### CODE REVIEW

**Comment by liatgrozovik** [comment]
hi marian

hope you enjoy and learn many things in the linux kernel course.

is this the fix only for single onie image,   right?
can you please elaborate what is the logic of the fix?

Best regards,
Liat Grozovik +972-54-923-7674

Sent from my Samsung Galaxy smartphone.


-------- Original message --------

File: files/build_templates/swss.service.j2 [hunk 1]
```diff
@@ -1,18 +1,40 @@
...
+ExecStartPre=/usr/bin/mst start
+ExecStartPre=/etc/mlnx/msn2700 start
+{% elif sonic_hwsku == 'AS7512' %}
+ExecStartPre=-/etc/init.d/xpnet.sh stop
+ExecStartPre=/etc/init.d/xpnet.sh start
+{% endif %}
+
+# systemd allows only one parent process within service,
+# so we spawn both dockers from single bash parent
+ExecStart=/bin/bash -c "/usr/bin/{{docker_container_name}}.sh start & /usr/bin/syncd.sh start"
```
Comment (lguohan) [inline files/build_templates/swss.service.j2]: what if we do ExecStartPre=/usr/bin/syncd.sh start?

what's the difference between your current solution?

**Comment by lguohan** [comment]
what is the linux kernel course? must be very interesting. 👍

File: files/build_templates/swss.service.j2 (hunk 1 above)
Comment (MaryanPritsak) [inline files/build_templates/swss.service.j2]: As I said in a comment, if we do syncd.sh start in ExecStartPre, systemd will wait until this process finishes and will never go reach ExecStart

Comment (qiluo-msft) [inline files/build_templates/swss.service.j2]: if you start first script in background, there is no way to get error code in bad situations.

---
In reply to: [103107923](https://github.com/Azure/sonic-buildimage/pull/334#discussion_r103107923) [](ancestors = 103107923)

Comment (lguohan) [inline files/build_templates/swss.service.j2]: @qiluo-msft, is there any way to resolve this?

**Comment by marian-pritsak** [comment]
Changed logic to return in case any of dockers exits with corresponding exit code

**Comment by qiluo-msft** [comment]
@marian-pritsak One more suggestion: if there are multiple iterations in the same pull request, please try your best to keep old commits that are reviewed, so all the discussion context will be kept.

**Pull request merged**